### DISCUSSION TITLE
条目录音提交和审核流程

### DISCUSSION THREAD

**Discussion by ztl8702** [MindongLab/yngdieng#204] [产品功能 ideas]
**你提出的功能想解决什么问题**

单字音的合成语音上线还需要一段时间。而连读音合成难度较大，也不是短时间内可以达到理想效果的。最理想读音的其实还是真人语音。因此考虑让网友提交读音，但是需要人工审核。

**你觉得理想的解决方案是什么**

* 榕典界面上提供一个“贡献语音”按钮；直接在网页上录音。
* 需要一个审核语音的后台界面。管理员可看到待审核的语音，试听，审核。若一个条目，有多个语音都通过审核，管理员可以为选择一个“最优”作为默认（点击“连读音”的时候播放），剩余的语音可显示在条目正文下方或隐藏。

**其他可能的解决方案**
你考虑过的，能解决同样问题的其他可能方案。

* 连读音 TTS ，理论上可实现，但执行起来肯定会有很多细节，不易做得“真”。可同步进行。
* 统一、分配请播音员、闽语演员等来录音。可同步进行。

**其他信息**

初步仅收集福州市区音。若行之有效，此系统可拓展成收集各地语音。

**Comment by ztl8702** [comment]
TODOs:

- [ ] 设计提交语音 UI
- [ ] 设计显示UI
- [ ] 设计审核流程和 UI 方案
- [ ] 技术设计

**Comment by Guanchishan** [comment]
> 初步仅收集福州市区音

关于网友自主提交音频，与其控制为福州市区音，不如一开始就广纳各闽东语读音？并允许上传者选择自己的区域（如仓山），填写自己的小区域（如黄山）。然后在后台审核上优先审核需要赶紧使用的“标准市区音”？

**Comment by ztl8702** [comment]
Good point

**Comment by ztl8702** [comment]
这个有点意思 https://www.appinn.com/xiangyin-home-speak/

**Comment by Guanchishan** [comment]
2020-08-15榕典/输入法例会上讨论认为，用户提交录音需要信息如下：

- 收集信息：性别/地方
- 收集信息时机：注册时填写

---

我们是否需要用户录音时的年龄？

如果用户注册时不填写出自何地，是否在每次上传时留下一个小选框鼓励用户把地址填了？

**Comment by Guanchishan** [comment]
界面设计稿@2020-08-17
![image](https://user-images.githubusercontent.com/30859354/90687900-2b75b300-e2a0-11ea-9428-4b36ab25f1d0.png)
图里有设计一个小磁贴纪念林圜先生。别的磁贴是点赞，林先生这个就改成点蜡，如图。
或许可以满足[对林先生特别致谢](https://github.com/MindongLab/yngdieng/issues/272#issuecomment-657645795)的需求。

**Comment by ztl8702** [comment]
我喜欢“用蜡烛代替点赞”的这个巧思，磁贴的视觉设计也很有意思，但是我有几个顾虑：

1. 既然是机器生成（虽然原始音频来自林先生），标注为“仓山”口音是否合适？比如说，我们用合成的语音发 `早 za33`(冯版大部分读音是“城里音”)，这样就不是仓山口音了？

2. 类似1的理由，我觉得也不适合在磁贴上直接写“林圜” -- 因为这不是林先生自己对这个词的念法。

**Comment by ztl8702** [comment]
“音韵”这个子页面的设计逻辑，可能还需要再调整，上图中，音韵页承载了太多功能：

* 不同人的发音（+ 社区化的反馈 aka 点赞按钮） -- 这个我觉得宜单独建一个“社区化发音”页面（名称待定），里面可以有更丰富的互动功能（如按地区筛选、贡献你的读音等）
* 单字分解 (供有兴趣进一步了解每个字的用户）-- 这个我觉得最符合“音韵”页的核心含义。
* 不同的拼音方案 
   * 这个像是另一个“维度”的功能。比如说，上面的每个不同发音人的发音，是否要提供不同书写/拼音系统的切换？还是说仅仅为钦定的读音提供不同拼音系统？（这样的话适合新建一个页面“拼音系统”）

------

**Comment by Guanchishan** [comment]
> 我觉得也不适合在磁贴上直接写“林圜” -- 因为这不是林先生自己对这个词的念法

嗯，最好想一个好办法来体现**只是声音来自他**，但是并不是他**对这个词的念法**。

Radium <notifications@github.com> 于2020年8月22日周六 下午10:09写道：

**Comment by Guanchishan** [comment]
> “音韵”这个子页面的设计逻辑，可能还需要再调整，上图中，音韵页承载了太多功能

我怕子页太多了，所以姑且都放一起，但是确实逻辑上他们是不一样的。是否放一起，放一起的话分页大标题怎么命名，都可以再讨论，反正不影响具体的功能设计。

> 仅仅为钦定的读音提供不同拼音系统？

倾向于后者。如果所有录音都提供转写，那磁贴是容不下的，除非为每个录音再开设子页面，但是这样的话次级页面又太多了（？）

**Comment by Guanchishan** [comment]
> 我喜欢“用蜡烛代替点赞”的这个巧思，磁贴的视觉设计也很有意思，但是我有几个顾虑：
> 
> 1. 既然是机器生成（虽然原始音频来自林先生），标注为“仓山”口音是否合适？比如说，我们用合成的语音发 `早 za33`(冯版大部分读音是“城里音”)，这样就不是仓山口音了？
> 2. 类似1的理由，我觉得也不适合在磁贴上直接写“林圜” -- 因为这不是林先生自己对这个词的念法。

我觉得林先生这张磁贴我们可以特别设计设计，和其他磁贴做得不一样。我之前满写多少岁、仓山，只是因为没想好怎么具体的设计此磁贴，所以满和其他的写成一样而已。
